### PR TITLE
CHE-467 Allow to configure Orion editor setting from IDE

### PR DESCRIPTION
**Pull request by RomanNikitenko**
@vparfonov  
editor_settings

### CODE REVIEW

File: core/ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/preference/editorproperties/EditorPropertiesManager.java [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+import static org.eclipse.che.ide.jseditor.client.preference.editorproperties.EditorProperties.SHOW_ZOOM_RULER;
+import static org.eclipse.che.ide.jseditor.client.preference.editorproperties.EditorProperties.SMART_INDENTATION;
+import static org.eclipse.che.ide.jseditor.client.preference.editorproperties.EditorProperties.TAB_SIZE;
+
+/**
+ * The class contains methods to simplify the work with editor properties.
+ *
+ * @author Roman Nikitenko
+ */
+
```
Comment (azatsarynnyy) [inline core/ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/preference/editorproperties/EditorPropertiesManager.java]: extra line

File: core/ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/preference/editorproperties/EditorPropertiesSectionPresenter.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+import com.google.web.bindery.event.shared.EventBus;
+
+import org.eclipse.che.api.machine.gwt.client.events.WsAgentStateEvent;
+import org.eclipse.che.api.machine.gwt.client.events.WsAgentStateHandler;
+import org.eclipse.che.ide.api.event.EditorSettingsChangedEvent;
+import org.eclipse.che.ide.jseditor.client.preference.EditorPreferenceSection;
+
+import java.util.Map;
+
+/** Presenter for the editor properties section in the 'Preferences' menu. */
```
Comment (azatsarynnyy) [inline core/ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/preference/editorproperties/EditorPropertiesSectionPresenter.java]: `author` tag skipped

File: core/ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/preference/editorproperties/EditorPropertiesSectionViewImpl.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+import java.util.Map;
+
+/**
+ * The class provides special panel to store and control editor's properties.
+ *
+ * @author Roman Nikitenko
+ */
+public class EditorPropertiesSectionViewImpl extends Composite implements EditorPropertiesSectionView, EditorPropertyWidget.ActionDelegate {
+
+    private static final CompilerSetupViewImplUiBinder UI_BINDER = GWT.create(CompilerSetupViewImplUiBinder.class);
```
Comment (azatsarynnyy) [inline core/ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/preference/editorproperties/EditorPropertiesSectionViewImpl.java]: Why `CompilerSetupViewImplUiBinder`?
`EditorPropertiesSectionViewImplUiBinder`

Comment (vparfonov) [inline core/ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/preference/editorproperties/EditorPropertiesSectionViewImpl.java]: +1

File: core/ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/preference/editorproperties/EditorPropertiesManager.java (hunk 1 above)
Comment (RomanNikitenko) [inline core/ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/preference/editorproperties/EditorPropertiesManager.java]: done

File: core/ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/preference/editorproperties/EditorPropertiesSectionPresenter.java (hunk 2 above)
Comment (RomanNikitenko) [inline core/ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/preference/editorproperties/EditorPropertiesSectionPresenter.java]: done

File: core/ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/preference/editorproperties/EditorPropertiesSectionViewImpl.java (hunk 3 above)
Comment (RomanNikitenko) [inline core/ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/preference/editorproperties/EditorPropertiesSectionViewImpl.java]: thank you, guys!
I've fixed it.

File: core/ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/preference/editorproperties/EditorPropertiesSectionPresenter.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    }
+
+    @Override
+    public void onWsAgentStarted(WsAgentStateEvent event) {
+        addEditorProperties();
+    }
+
+    @Override
+    public void onWsAgentStopped(WsAgentStateEvent event) {
+        //do nothing
```
Comment (azatsarynnyy) [inline core/ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/preference/editorproperties/EditorPropertiesSectionPresenter.java]: Is it really useful comment? :smile:

Comment (RomanNikitenko) [inline core/ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/preference/editorproperties/EditorPropertiesSectionPresenter.java]: This comment means for me: "I didn't forget to write some logic at this place - we don't need some logic here"
But up to you - I can delete it :-)

Comment (RomanNikitenko) [inline core/ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/preference/editorproperties/EditorPropertiesSectionPresenter.java]: Done

Comment (azatsarynnyy) [inline core/ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/preference/editorproperties/EditorPropertiesSectionPresenter.java]: I don't insist. Just was intrested

File: core/ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/event/EditorSettingsChangedEvent.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+package org.eclipse.che.ide.api.event;
+
+import com.google.gwt.event.shared.GwtEvent;
+
+/**
+ * Event fired when editor's settings has been changed.
+ *
+ * @author Roman Nikitenko
+ */
+public class EditorSettingsChangedEvent extends GwtEvent<EditorSettingsChangedHandler> {
```
Comment (vzhukovs) [inline core/ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/event/EditorSettingsChangedEvent.java]: FYI, you can combine event and handler into one class, as it did here https://github.com/eclipse/che/blob/projectManager/core/ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/resources/ResourceChangedEvent.java

It is easier for maintaining in the future.

File: core/ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/preference/editorproperties/EditorProperties.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+import javax.validation.constraints.NotNull;
+
+/**
+ * The interface contains editor's constants.
+ *
+ * @author Roman Nikitenko
+ */
+public enum EditorProperties {
+
+    TAB_SIZE("tabSize"),
```
Comment (vzhukovs) [inline core/ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/preference/editorproperties/EditorProperties.java:23]: Can you add also property for showing white spaces?

Correspondent property name is `showWhitespaces`.

Comment (RomanNikitenko) [inline core/ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/preference/editorproperties/EditorProperties.java:23]: done

File: core/ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/event/EditorSettingsChangedEvent.java (hunk 5 above)
Comment (RomanNikitenko) [inline core/ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/event/EditorSettingsChangedEvent.java]: done, thank you!

**Comment by RomanNikitenko** [comment]
@slemeur @vparfonov I've updated UI for editor settings in the preference-menu

**Comment by slemeur** [comment]
OK for me

**Comment by vparfonov** [comment]
ok

**Pull request merged**